tests/main/snap-session-agent-unavailable-to-snaps/task.yaml

```diff
@@ -0,0 +1,55 @@
+summary: Check that snaps can not access the session agent
+
+systems:
+    # Ubuntu 14.04 does not have a complete systemd implementation
+    - -ubuntu-14.04-*
+    # Systemd on CentOS 7/Amazon Linux 2 does not have the user@uid unit
+    - -amazon-linux-2-*
+    - -centos-7-*
+
+environment:
+    TEST_UID: $(id -u test)
+    USER_RUNTIME_DIR: /run/user/${TEST_UID}
+
+prepare: |
+    # Ensure that snapd.session-agent.socket is enabled.  This may not
+    # be the case on distributions where presets have been used to
+    # disable it.
+    if [ ! -L /usr/lib/systemd/user/sockets.target.wants/snapd.session-agent.socket ] &&
+            ! systemctl --user --global is-enabled snapd.session-agent.socket; then
+        systemctl --user --global enable snapd.session-agent.socket
+        touch agent-was-enabled
+    fi
+    mkdir -p "$USER_RUNTIME_DIR"
+    chmod u=rwX,go= "$USER_RUNTIME_DIR"
+    chown test:test "$USER_RUNTIME_DIR"
+    systemctl start "user@${TEST_UID}.service"
+
+restore: |
+    systemctl stop "user@${TEST_UID}.service"
+    rm -rf "${USER_RUNTIME_DIR:?}"/* "${USER_RUNTIME_DIR:?}"/.[!.]*
+    if [ -f agent-was-enabled ]; then
+        systemctl --user --global disable snapd.session-agent.socket
+        rm agent-was-enabled
```

Comment:
no need to delete this file, the test directory is automatically cleaned during restore

Reply:
Are you sure about this?  I can see code to try and restore system level units, but nothing to change the enable/disable state for use level units.

Reply:
I'm not sure how the remark and the question relates.  @sergiocazzolato's remark was only about the cleaning up of files in the test dir, nothing to do about services or units

Reply:
At the same time, cleaning up the files does not really hurt.